### PR TITLE
Add currency notation

### PR DESCRIPTION
**Pull request by GreaseMonk**
## About the PR

This PR changes the display of spesos amounts from  $1000000  to  $1,000,000 (depending on the CultureInfo).

## Why / Balance

Makes it easier to discern

## How to test

Check the bank, vending machines, PDA, etc.

## Media

image
image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Currency notation

### CODE REVIEW

**Comment by arimah** [comment]
No, you will not use `.` as a thousands separator in an English-language game, because `.` is the decimal separator in almost all English-speaking regions. You will use the locale-specific number format, such that it turns into `,` when the client is in English, ` ` when it's in Russian, and so on. And you will use the built-in locale formatting such that the Hindi server created five years from now uses correct placement of digit grouping for that region (`12,34,567.89` – yes, only the least significant group is three digits).

**Comment by arimah** [comment]
You will also not hardcode the position of the currency symbol nor the currency symbol itself. In many parts of the world, the currency symbol comes _after_ the amount, ex. in Russian: `1 234 567,89 $`

**Comment by dvir001** [comment]
Like the PR but it needes the ``,``

And please add support to add prefix and suffix options that are not hardcoded

File: Resources/Prototypes/_NF/Catalog/Fills/Items/briefcases.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@
 - type: entity
   parent: SpaceCashExpedition
   id: SpaceCashMinimalExpedition
-  name: spesos (5000)
+  name: spesos (5.000)
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Catalog/Fills/Items/briefcases.yml:21]: Consistency with en-US cultureinfo.
```suggestion
  name: spesos (5,000)
```

File: Resources/Prototypes/_NF/Catalog/Fills/Items/briefcases.yml [hunk 2]
```diff
@@ -32,7 +32,7 @@
 - type: entity
   parent: SpaceCashExpedition
   id: SpaceCashMinorExpedition
-  name: spesos (10000)
+  name: spesos (10.000)
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Catalog/Fills/Items/briefcases.yml:35]: Consistency with en-US cultureinfo.
```suggestion
  name: spesos (10,000)
```

File: Resources/Prototypes/_NF/Catalog/Fills/Items/briefcases.yml [hunk 3]
```diff
@@ -46,7 +46,7 @@
 - type: entity
   parent: SpaceCashExpedition
   id: SpaceCashModerateExpedition
-  name: spesos (20000)
+  name: spesos (20.000)
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Catalog/Fills/Items/briefcases.yml:49]: Consistency with en-US cultureinfo.
```suggestion
  name: spesos (20,000)
```

File: Resources/Prototypes/_NF/Catalog/Fills/Items/briefcases.yml [hunk 4]
```diff
@@ -60,7 +60,7 @@
 - type: entity
   parent: SpaceCashExpedition
   id: SpaceCashHazardousExpedition
-  name: spesos (40000)
+  name: spesos (40.000)
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Catalog/Fills/Items/briefcases.yml:63]: Consistency with en-US cultureinfo.
```suggestion
  name: spesos (40,000)
```

File: Resources/Prototypes/_NF/Catalog/Fills/Items/briefcases.yml [hunk 5]
```diff
@@ -75,7 +75,7 @@
 - type: entity
   parent: SpaceCashExpedition
   id: SpaceCashExtremeExpedition
-  name: spesos (70000)
+  name: spesos (70.000)
```
Comment (whatston3) [inline Resources/Prototypes/_NF/Catalog/Fills/Items/briefcases.yml:78]: Consistency with en-US cultureinfo.
```suggestion
  name: spesos (70,000)
```

**Review by whatston3 — CHANGES_REQUESTED** [review]
The biggest issue I have with this is that some entries, in particular, YAML or FTL definitions for papers (see #1728) do not have a good way to inject this, so you have inconsistently localized oddities.

Apart from that, looks alright.  I've raised concerns where I have them in the code.  Will test and report back.

**Comment by whatston3** [comment]
Testing seemed fine.

**Comment by whatston3** [comment]
Pushed a set of suggestions up onto 2024-08-03-balance-suggestions on my fork.

[Diff is available here (GitHub)](https://github.com/GreaseMonk/frontier-station-14/compare/balance...whatston3:frontier-station-14:2024-08-03-balance-suggestions?expand=1).

This allows the functions to be used for DB, uses $ specifically for spesos, and creates a set of convenience methods to get values in a particular currency, or to get a value independent of currency (function name for that one sucks, sorry).

I'd be happy to approve the branch with these changes in, and might suggest checking lines over for which are missing comments.

Here's a picture of these functions in use with the pirate uplink.

doubloons

I don't really endorse this feature, seems like it touches many things for little gain, but with this, we can at least use it consistently.

**Review by whatston3 — APPROVED** [review]

**Pull request merged**